build-tools/packages/build-cli/src/library/repoPolicyCheck/fluidBuildTasks.ts

```diff
@@ -301,30 +231,64 @@ function eslintGetScriptDependencies(
 		throw new Error(`Unable to load eslint config file ${eslintConfig}. ${error}`);
 	}
 
-	let projects: string | string[] | undefined = config.parserOptions?.project;
-	if (projects === undefined) {
-		// If we don't have projects, our task needs to have dependent build scripts
-		return getDefaultTscTaskDependencies(root, json);
+	let projects = config.parserOptions?.project;
+	if (!Array.isArray(projects) && typeof projects !== "string") {
+		// config does not parse extends. So, "project" is what is set in top file.
+		if (projects === false || projects === null) {
+			// type based linting is disabled - assume no task prerequisites
+			return [];
+		}
+		// @typescript-eslint/parser allows true to mean use closest tsconfig.json, but we want
+		// explicit listings for dependency clarity.
+		if (projects === true) {
+			throw new Error(
+				`${json.name} eslint config 'project' is true, but should be list of tsconfig files.`,
+			);
+		}
+		// It undefined to leverage the expected base config setting of ./tsconfig.json.
```

Comment:
Did I get this one right?
```suggestion
		// projects === undefined, meaning "use the expected base config setting of ./tsconfig.json".
```